lib/CustomFields/ApiDataSanitizer.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace MailPoet\CustomFields;
+
+use InvalidArgumentException;
+use MailPoet\Models\CustomField;
+
+class ApiDataSanitizer {
+
+  function sanitize(array $data = []) {
+    $this->checkMandatoryStringParameter($data, 'name');
+    $this->checkMandatoryStringParameter($data, 'type');
+    $this->checkParamsType($data);
+    return [
+      'name' => $data['name'],
+      'type' => strtolower($data['type']),
+      'params' => $this->sanitizeParams($data),
+    ];
+  }
+
+  private function checkMandatoryStringParameter(array $data, $parameter_name) {
+    if (!isset($data[$parameter_name]) || empty($data[$parameter_name])) {
```

Comment:
Only the `empty(...)` part would be enough here: https://3v4l.org/TihcG

Reply:
But maybe here (and in other cases where `empty` is used) it should be considered that even some non-empty value are considered empty by this function (like string `'0'`) - https://www.php.net/empty. It's really an edge case thought so probably not much to worry about.